egs/librispeech/ASR/conformer_lm/dataset.py

```diff
@@ -698,8 +712,20 @@ class does not retain a reference to the LmDataset.
         # `data_indexes` above (this is not stored, as we know the formula).
         self.sentence_lengths = sentence_lengths
 
-        self.set_epoch(0)  # this is responsible for setting self.sorted_data_indexes
-
+        if not delay_init:
+            self.set_epoch(0)  # this is responsible for setting self.sorted_data_indexes
+
+    def _sync_sizes(self, device: torch.device = torch.device('cuda')):
+        # Calling this on all copies of a DDP setup will sync the sizes so that
+        # all copies have the exact same number of batches.  I think
```

Comment:
Shall we mention that without doing this, the training process 
will hang indefinitely at the end?

Reply:
Mm, sure...